trio/_dtls.py

```diff
@@ -1207,13 +1267,17 @@ def __exit__(
     ) -> None:
         return self.close()
 
-    def _check_closed(self):
+    def _check_closed(self) -> None:
         if self._closed:
             raise trio.ClosedResourceError
 
     async def serve(
-        self, ssl_context, async_fn, *args, task_status=trio.TASK_STATUS_IGNORED
-    ):
+        self,
+        ssl_context: Context,
+        async_fn: Callable[[DTLSChannel], Awaitable],
+        *args: Any,
```

Comment:
It looks like `*args` are positional arguments to `async_fn` which are passed after the `stream: DTLSChannel`, meaning that our type annotation for `async_fn` is incorrect to specify that it must have only one argument - and for `*args` is weaker than we could manage with `ParamSpec` and `Concatenate`.

I think we should _either_ weaken these annotations to `async_fn: Callable[..., Awaitable]`, or strengthen them to `async_fn: Callable[Concatenate[DTLSChannel, P], Awaitable], *args: P.args`.  I'd prefer the latter if it's feasible given deps and python versions.

Reply:
`ParamSpec` requires using both `*args` and `*kwargs`, only one doesn't work. For typing this we need `TypeVarTuple`, but support for that in Mypy is currently experimental and incomplete.

Reply:
😭 

Alright, let's use the simpler annotation then and leave a comment about what we want to do and that it's blocked on Mypy supporting `TypeVarTuple`.

Reply:
Or we could extend the API to accept `*kwargs` - though `pyright` requires all other parameters to be before both `P.args` and `P.kwargs` so that would ultimately break stuff.

Weakening the annotation and adding a comment :+1:

Reply:
No kwargs; they make it impossible to backwards-compatibly add new options to the function later.  We use the allow-args/use-partial-for-kwargs style in many places.